cpp/include/cudf/lists/drop_list_duplicates.hpp

```diff
@@ -28,35 +28,103 @@ namespace lists {
  */
 
 /**
- * @brief Create a new lists column by extracting unique entries from list elements in the given
- * lists column.
- *
- * Given an input lists column, the list elements in the column are copied to an output lists
- * column such that their duplicated entries are dropped out to keep only the unique ones. The
- * order of those entries within each list are not guaranteed to be preserved as in the input. In
- * the current implementation, entries in the output lists are sorted by ascending order (nulls
- * last), but this is not guaranteed in future implementation.
- *
- * @throw cudf::logic_error if the child column of the input lists column contains nested type other
- * than struct.
- *
- * @param lists_column The input lists column to extract lists with unique entries.
- * @param nulls_equal Flag to specify whether null entries should be considered equal.
- * @param nans_equal Flag to specify whether NaN entries should be considered as equal value (only
- *        applicable for floating point data column).
+ * @brief Create new lists columns by copying entries from the input lists columns ignoring
+ * duplicate list entries.
+ *
+ * Given a pair of keys-values lists columns where both columns have the same number of rows
+ * and the size of each corresponding row is the same, remove entries from both columns where the
+ * corresponding key values are the same.
+ *
+ * A pair of output keys and values lists columns are generated by copying entries from the input
+ * lists columns. The duplicate entries in each list of the keys column are ignored during copying
+ * to output only the unique list entries. Entries in the values column mapped to these key entries
+ * are also copied or ignored along with them.
+ *
+ * The users are responsible for the keys-values lists columns having the same number of entries
+ * at each row. Otherwise, a crash may happen, or the results will be undefined.
```

Comment:
```suggestion
 * The users are responsible for the keys-values lists columns having the same number of entries
 * at each row. Otherwise, behavior is undefined. 
```